plugins/woocommerce/src/Internal/Admin/Orders/Edit.php

```diff
@@ -155,24 +156,28 @@ public function setup( \WC_Order $order ) {
 		$this->add_order_specific_meta_box();
 		$this->add_order_taxonomies_meta_box();
 
-		/**
-		 * From wp-admin/includes/meta-boxes.php.
-		 *
-		 * Fires after all built-in meta boxes have been added. Custom metaboxes may be enqueued here.
-		 *
-		 * @since 3.8.0.
-		 */
-		do_action( 'add_meta_boxes', $this->screen_id, $this->order );
+		$order_post = get_post( $this->order->get_id() );
 
-		/**
-		 * Provides an opportunity to inject custom meta boxes into the order editor screen. This
-		 * hook is an analog of `add_meta_boxes_<POST_TYPE>` as provided by WordPress core.
-		 *
-		 * @since 7.4.0
-		 *
-		 * @param WC_Order $order The order being edited.
-		 */
-		do_action( 'add_meta_boxes_' . $this->screen_id, $this->order );
+		if ( $order_post instanceof WP_Post ) {
```

Comment:
Thanks for submitting this. 

We'll discuss it, but I don't think we can accept the PR in its current form because, in reality and given we're inside the HPOS order editor code, I don't think this check would ever be satisfied. 

Though I understand the problem it seeks to solve, this will cause a different set of breakages and back-compat problems that your testing instructions don't cover (for example, WooCommerce extensions that anticipate the current behavior and rely on it to integrate with the order editor).

Reply:
Hi @barryhughes, Thank you for your reply.

> I don't think this check would ever be satisfied.

It actually does satisfy, the reason is that even with HPOS, a placeholder entry is created in the `wp_posts` table (please [see here](https://github.com/woocommerce/woocommerce/blob/35ba49a1f29bad8a1685ae4a179c7273eee3f298/docs/high-performance-order-storage/README.md#placeholder-records)).

> Though I understand the problem it seeks to solve, this will cause a different set of breakages and back-compat problems that your testing instructions don't cover (for example, WooCommerce extensions that anticipate the current behavior and rely on it to integrate with the order editor).

I understand your concern about other plugins using the same hook, but objects of `Order` are unexpected as second parameter of `add_meta_boxes` hook, according to the [hook reference](https://developer.wordpress.org/reference/hooks/add_meta_boxes/).

So even for the "WooCommerce extensions that anticipate the current behavior" this issue will cause a fatal error as soon as they use a native type hint (because other WordPress metaboxes are passing `WP_Post`).